src/lib.rs

```diff
@@ -212,56 +242,62 @@ impl<O> IO<O> {
 /// assert_eq!(result, "meow\n");
 /// # Ok (()) }
 /// ```
-pub fn pure <T> (value: T) -> IO<T> {
+pub fn pure <'a, T> (value: T) -> IO<'a, T> where T: 'a {
     IO (Pure (value))
 }
 
 /// Ignore the output value of `x`.
-pub fn void <T> (x: IO<T>) -> IO<()>
+pub fn void <'a, T> (x: IO<'a, T>) -> IO<'a>
```

Comment:
I think the docs for this function could be improved? As a non FP dev this makes me guess that it either: "Does this make IO evaluate? Or is this equivalent to `.map(|_| ())`?" I think the second one is right but I don't really know

Reply:
These docs are so great, clean and understandable, way better what I had in mind 👍. I love them